### ISSUE TITLE
Settings:SelectBox: Add aria-describedby for extended descriptions

### ISSUE THREAD

**Issue by cleidigh** [Microsoft/vscode#57710]
New enum/select box detailed descriptions should be read by reader.
Add describedby attribute handling for details provider within selectbox.

**Comment by cleidigh** [comment]
Yep, I've thought about live/alert also.  
By my research this does not really match the expected taxonomy for the accessibility tree,
an extended description is pretty much exactly what describedby is for.
That said, describedby probably will not be happy changing which will mean I will need
the static set of descriptions in the Dom.  Not a big deal, just another example of where
nothing is straightforward.  But, that's why we get paid the big bucks ... well at least you :-o !

What I really want to solve is getting rid of the multiple 'clickable' readings.  I think I may have a way to avoid that which will be very important for these descriptions given the number of links.

**Comment by roblourens** [comment]
> That said, describedby probably will not be happy changing which will mean I will need
> the static set of descriptions in the Dom.

Good point, I have no idea if pointing all of the selectbox items at the same details box will work, or if there's something we can do to order events to make it work better. Or maybe changing the 'id' of the details element for each item.

> What I really want to solve is getting rid of the multiple 'clickable' readings

I'm very curious what's causing that! Very weird.

**Comment by cleidigh** [comment]
describedby can have a list of IDs associated with it.  like labeled by and most other ARIA aspects
it does not like things changing underneath it, that of course is what aria-live is for but, but everything I read says use this judiciously.  I think having a separate set of descriptions (modified for reader output/minimal clickable) will not be a big deal.  Just as the options are static, so are the descriptions
so hiding them in the select for use will not be an issue.

I haven't done the definitive experiment yet but, I believe the multiple clickable issue has to do with something like this:

<div>
<p>this is an extended description <a>link</a></p>
</div>

I think the reader can find these as all clickable elements when they are descendents of a managed
accessibility tree.  my research has yet to determine if this is considered a bug or as designed
